### ISSUE TITLE
Output contract metadata in build artifacts

### ISSUE THREAD

**Issue by kuzdogan** [eth-brownie/brownie#1591]
### Overview

Recently I got my hands on [a project](https://github.com/beamer-bridge/beamer) that compiles contracts with brownie and wants to verify their contracts with [Sourcify](https://github.com/ethereum/sourcify). Sourcify does [full/perfect verification](https://docs.sourcify.dev/docs/full-vs-partial-match/) which leverages Solidity's [contract metadata](https://docs.soliditylang.org/en/latest/metadata.html) and the hash of it appended at the end of the bytecode. 

To be able to perfectly verify contracts, the contract metadata is needed and brownie doesn't seem to output this information in the contract's compilation artifacts.

### Specification

To be able to verify, users need to provide the contract metadata alongside the source files. In brownie, I noticed the metadata is not output in the `build/contracts/*.json` files by default. In contrast, metadata is output by default in Truffle, and since v2.6.8 in Hardhat. 

I also couldn't find an option to modify the [Standard JSON Input-Output](https://docs.soliditylang.org/en/latest/using-the-compiler.html#compiler-input-and-output-json-description) of the compiler where it would be possible to output the metadata as such:
```
version: ...,
  optimizer: {...},
  outputSelection: {
    "*": {
      "*": [
        "abi",
        "evm.bytecode",
        "evm.deployedBytecode",
        "metadata", // <-- add this
        ...
      ]
    },
  },
```

Including a Truffle output file as a reference. Notice the metadata field:
[Resolver.json.txt](https://github.com/eth-brownie/brownie/files/9269206/Resolver.json.txt)

Also if the metadata file becomes available, it would be possible to develop a Sourcify verification plugin to easily verify contracts following the deployment of the contract.

cc: @fredo

**Comment by Jeevan-J** [comment]
I agree with exporting the contract metadata information in build folder is needed for further verification manually. But `publish_source` argument in deploy function does the automatic verification, right?

**Comment by kuzdogan** [comment]
@Jeevan-J Yes it seems, but it verifies on Etherscan and Etherscan verification does not require the metadata file. If you're interested [this article](https://docs.sourcify.dev/blog/verify-contracts-perfectly/) explains the difference in detail

**Comment by istankovic** [comment]
Thanks @kuzdogan for submitting this!
Yeah, this is something that would be really nice to have.

@Jeevan-J are you a maintainer of eth-brownie? Would you accept patches for this?

**Comment by marcocastignoli** [comment]
There is this open PR: https://github.com/eth-brownie/brownie/pull/1280
